shell/browser/ui/cocoa/electron_menu_controller.mm

```diff
@@ -321,6 +321,15 @@ - (NSMenuItem*)makeMenuItemForIndex:(NSInteger)index
                                                 action:@selector(itemSelected:)
                                          keyEquivalent:@""];
 
+  std::u16string role = model->GetRoleAt(index);
+  electron::ElectronMenuModel::ItemType type = model->GetTypeAt(index);
+
+  if (@available(macOS 14, *)) {
```

Comment:
It's probably worth documenting that the `header` role is only available in macOS 14+

Reply:
[`cecf5f5`](https://github.com/electron/electron/pull/45538/commits/cecf5f565604b466892b5ff49ff4d385bab8f1e5)